containers/python-3-anaconda-postgres/test-project/hello.py

```diff
@@ -0,0 +1,54 @@
+#-------------------------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See https://go.microsoft.com/fwlink/?linkid=2090316 for license information.
+#-------------------------------------------------------------------------------------------------------------
+
+#%%
+import os
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+import psycopg2
+
+try:
+    pg_user = os.getenv('POSTGRES_USER')
+    pg_password = os.getenv('POSTGRES_PASSWORD')
+    pg_db = os.getenv('POSTGRES_DB')
+    pg_host = os.getenv('POSTGRES_HOST')
+    conn = psycopg2.connect("dbname='{pg_db}' user='{pg_user}' host='{pg_host}' password='{pg_password}'".format(
+        pg_db=pg_db,
+        pg_user=pg_user,
+        pg_host=pg_host,
+        pg_password=pg_password
+    ))
+except:
+    print("Unable to connect to the database")
+    exit()
+
+try:
+    with conn.cursor() as cur:
+        cur.execute("""SELECT COUNT(1) from pg_database WHERE datname='postgres'""")
+        rows = cur.fetchone()
+
+    if rows[0]==1:
+        print("DATABASE CONNECTED")
+    else:
+        print("ERROR FINDING DATABASE")
+finally:
+    conn.close()
+
+# # Data for plotting
+t = np.arange(0.0, 2.0, 0.01)
```

Comment:
Are these part of the test? Not sure if the goal is to make sure the db is up and running only or something else too.

Reply:
The goal is to exercise the main use case for this container which would be connecting to a db and then using that data to do some Data Science ™️ (in this case using matplotlib to make a chart).  So this script does both (although, since it's all made up those two pieces are unrelated to each other)

Reply:
My suggestion would be to split into different tests, using some test framework (i.e pytest) so it would be more clear and easily to understand and maintain. Having a clear deliniation between tests would help.

Reply:
I understand where this suggestion is coming from but the script is not intended to be a test script.  Rather, it is meant to be a data processing script (which is the most common usage of anaconda).

I think there is a separate argument as to whether or not unit tests are appropriate for this example.  I will do a little digging and see if I can find some reasonable examples for testing a script like this, but I in my experience they are generally not unit tested